### ISSUE TITLE
Use Flow type information, e.g. for require-parameter-type

### ISSUE THREAD

**Issue by mrtnbroder** [gajus/eslint-plugin-flowtype#124]
I've started to write a type definition for the node [nock](https://github.com/node-nock/nock) library, which currently looks like this:

``` js
declare module nock {

  declare type ReplyCallback = (uri: string, body: any) => [number, mixed]

  declare type KeyValue = { [key: any]: any }

  declare class Nock {
    (url: string): Nock;

    cleanAll: () => Nock;
    get: (url: string) => Nock,
    put: (url: string) => Nock,
    post: (url: string, opts: any) => Nock,
    get: (url: string) => Nock,
    delete: (url: string) => Nock,
    reply: {
      (statusCode: number, opts?: KeyValue): Nock,
      (fn: ReplyCallback): Nock,
    },
    isDone: () => boolean,
    query: (query: KeyValue | Function) => Nock,
  }

  declare export default Nock
}
```

Given the following test example:

``` js
nock('http://www.example.com')
    .get('/dogs')
    .reply((uri) => { // warning: missing parameter type for `uri`
      return [200, 'done']
    })
```

It's complaining within the callback function about the missing type for `uri`, though - it's there, within the module definition.

Imo, `require-parameter-type` shouldn't complain here, as it **does** have a type definition, even though further down the tree.

**Comment by danharper** [comment]
I think ESLint has future plans to support communication with external services, which would allow us to lint assisted by Flow's inferred type information. Until then, we can really only lint against information declared on the node.

So for now, this plugin is for literally "add type annotations _everywhere_", which can be pretty overkill.

**Comment by mrtnbroder** [comment]
_sigh_, I see, thanks for the info @danharper.

Leaving this up to you to close this issue then!

**Comment by danharper** [comment]
I've renamed the title a little, and I'll leave it open as a future reminder 👍

**Comment by gajus** [comment]
Closing as stale.
